sematic/resolvers/state_machine_resolver.py

```diff
@@ -402,3 +415,114 @@ def _set_nested_future(
     @classmethod
     def classpath(cls) -> str:
         return f"{cls.__module__}.{cls.__name__}"
+
+    @classmethod
+    def _get_resource_manager(cls) -> AbstractResourceManager:
+        raise NotImplementedError("Child classes must implement _get_resource_manager")
+
+    @classmethod
+    def _do_resource_activate(
+        cls, resource: AbstractExternalResource
+    ) -> AbstractExternalResource:
+        raise NotImplementedError("Child classes must implement _do_resource_activate")
+
+    @classmethod
+    def _do_resource_deactivate(
+        cls, resource: AbstractExternalResource
+    ) -> AbstractExternalResource:
+        raise NotImplementedError(
+            "Child classes must implement _do_resource_deactivate"
+        )
+
+    @classmethod
+    def _do_resource_update(
+        cls, resource: AbstractExternalResource
+    ) -> AbstractExternalResource:
+        raise NotImplementedError("Child classes must implement _do_resource_update")
+
+    @classmethod
+    def _save_resource(cls, resource: AbstractExternalResource):
+        raise NotImplementedError("Child classes must implement _save_resource")
+
+    @classmethod
+    def activate_resource_for_run(  # type: ignore
```

Comment:
Please add debug logs to `activate_resource_for_run` and to `deactivate_resource` that these operations are commencing, and air out the code.